### ISSUE TITLE
free-bits implementation is different from DreamerV3

### ISSUE THREAD

**Issue by 2M-kotb** [weipu-zhang/STORM#12]
Hi,

Congratulations on the great work, and thank you for sharing the code.

I notice that your implementation of free-bits is after performing a mean operation on KL loss as follows:

```
kl_div = kl_div.mean()
kl_div = torch.max(torch.ones_like(kl_div)*self.free_bits, kl_div)
```

While in DreamerV3, free-bits is performed before the mean operation as follows:

```
rep_loss = torch.clip(rep_loss, min=free)
dyn_loss = torch.clip(dyn_loss, min=free)
```

I wonder, what is the difference? I guess you perform some sort of a global free-bits, while DreamerV3 perform a per-class free bits

**Comment by weipu-zhang** [comment]
Hi,

Since there's no official PyTorch implementation for DreamerV3, I referred to the JAX version when implementing this part, though I might be misremembering the exact details. You can check the JAX implementation if interested.

That said, in my experiments, this difference didn't seem to have a significant impact on the final performance.

**Comment by 2M-kotb** [comment]
Thank you for the response. I checked the JAX implementation and there is no much difference.
